src/catchup/VerifyLedgerChainWork.cpp

```diff
@@ -284,29 +308,20 @@ VerifyLedgerChainWork::verifyAgainstTrustedHash(
 Work::State
 VerifyLedgerChainWork::onSuccess()
 {
-    mApp.getCatchupManager().logAndUpdateCatchupStatus(true);
-
-    if (mCurrCheckpoint <
-        mApp.getHistoryManager().checkpointContainingLedger(mRange.first()))
+    if (!mReady)
     {
-        throw std::runtime_error(
-            "Verification undershot first ledger in the range.");
+        return WORK_PENDING;
```

Comment:
I think there is a logical bug here:
if we're not ready, why did we schedule success? We should continue running instead